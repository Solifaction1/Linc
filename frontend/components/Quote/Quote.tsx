import React from 'react'
import QuoteData from '@models/quote.model'
import { defaultProps } from './quote.config'
import styles from './quote.module.scss'

const Quote: React.FC<QuoteData> = (props: QuoteData) => (
	<section className={styles.quote}>
		<div className={styles.title}>{props.title}</div>

		{props.employee.data && (
			<p className={styles.author}>
				{`${props.employee.data.attributes.firstName} ${
					props.employee.data.attributes.name
				}${
					props.employee.data.attributes.role !== null
						? ` - ${props.employee.data.attributes.role}`
						: ''
				}${
					props.employee.data.attributes.rechtsdomeinen.data.length > 0
						? props.employee.data.attributes.rechtsdomeinen.data.map(
								(rechtsdomein) => ` - ${rechtsdomein.attributes.title}`
						  )
						: ''
				}`}
			</p>
		)}
	</section>
)

Quote.defaultProps = defaultProps

export default Quote
