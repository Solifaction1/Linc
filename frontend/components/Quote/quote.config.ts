/* eslint-disable max-lines */
import QuoteData from "@models/quote.model"

const defaultProps: QuoteData = {
	id: 2,
	title: 'title',
	employee: {
		data: {
			id: 0,
			attributes: {
				name: 'name',
				locale: 'locale',
				firstName: 'firstName',
				role: 'role',
				rechtsdomeinen: {
					data: [{
						id: 1,
						attributes: {
							title: 'title',
							description: {
								text: '<p>test</p>'
							},
							image: {
								data: null
							},
							slug: 'slug',
							updatedAt: 'new Date()',
							publishedAt: 'new Date()',
							createdAt: 'new Date()',
							component: [],
							locale: 'locale',
							seo: null,
							body: []
						}
					}]
				},
				telephone: 'telephone',
				slug: 'slug',
				updatedAt: 'new Date()',
				publishedAt: 'new Date()',
				createdAt: 'new Date()',
				color: 'color',
				header: {
					color: 'gray',
					imagePosition: 'imagePosition',
					image: { data: null },
					CTA: null,
					bottomText: { cta: null }
				},
				linkedin: null,
				showInNavigation: true
			}
		}
	}
}

const defaultPropsNoRole: QuoteData = {
	id: 2,
	title: 'title',
	employee: {
		data: {
			id: 0,
			attributes: {
				name: 'name',
				locale: 'locale',
				firstName: 'firstName',
				role: null,
				rechtsdomeinen: {
					data: [{
						id: 1,
						attributes: {
							title: 'title',
							description: {
								text: '<p>test</p>'
							},
							image: {
								data: null
							},
							slug: 'slug',
							updatedAt: 'new Date()',
							publishedAt: 'new Date()',
							createdAt: 'new Date()',
							component: [],
							locale: 'locale',
							seo: null,
							body: []
						}
					}]
				},
				telephone: 'telephone',
				slug: 'slug',
				updatedAt: 'new Date()',
				publishedAt: 'new Date()',
				createdAt: 'new Date()',
				color: 'color',
				header: {
					color: 'gray',
					imagePosition: 'imagePosition',
					image: { data: null },
					CTA: null,
					bottomText: { cta: null }
				},
				linkedin: null,
				showInNavigation: true
			}
		}
	}
}

const defaultPropsNoRoleAndNoRechtsDomein: QuoteData = {
	id: 2,
	title: 'title',
	employee: {
		data: {
			id: 0,
			attributes: {
				name: 'name',
				locale: 'locale',
				firstName: 'firstName',
				role: null,
				rechtsdomeinen: {
					data: []
				},
				telephone: 'telephone',
				slug: 'slug',
				updatedAt: 'new Date()',
				publishedAt: 'new Date()',
				createdAt: 'new Date()',
				color: 'color',
				header: {
					color: 'gray',
					imagePosition: 'imagePosition',
					image: { data: null },
					CTA: null,
					bottomText: { cta: null }
				},
				linkedin: null,
				showInNavigation: true
			}
		}
	}
}

const defaultPropsNoRechtsDomein: QuoteData = {
	id: 2,
	title: 'title',
	employee: {
		data: {
			id: 0,
			attributes: {
				name: 'name',
				locale: 'locale',
				firstName: 'firstName',
				role: 'role',
				rechtsdomeinen: {
					data: []
				},
				telephone: 'telephone',
				slug: 'slug',
				updatedAt: 'new Date()',
				publishedAt: 'new Date()',
				createdAt: 'new Date()',
				color: 'color',
				header: {
					color: 'gray',
					imagePosition: 'imagePosition',
					image: { data: null },
					CTA: null,
					bottomText: { cta: null }
				},
				linkedin: null,
				showInNavigation: true
			}
		}
	}
}

export { defaultProps, defaultPropsNoRole, defaultPropsNoRoleAndNoRechtsDomein, defaultPropsNoRechtsDomein }
