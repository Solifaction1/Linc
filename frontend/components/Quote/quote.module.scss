@import 'styles/vars';
@import 'styles/utilities';

.quote {
	@extend %page-container;
	grid-template-areas:
		'. title title title title title title title title title title .'
		'author author author author author author author author author author author author';
	gap: calc(#{$spacing-base} * 2);

	min-height: 30rem;
	margin-top: calc(#{$spacing-base} * 7.5);
	margin-bottom: calc(#{$spacing-base} * 7.5);
	padding-top: calc(#{$spacing-base} * 7.5);
	padding-bottom: calc(#{$spacing-base} * 7.5);

	text-align: center;

	@media (max-width: $break-md) {
		margin-top: calc(#{$spacing-base} * 4);
		margin-bottom: calc(#{$spacing-base} * 4);
		padding-top: calc(#{$spacing-base} * 4);
		padding-bottom: calc(#{$spacing-base} * 4);
	}

	.title {
		grid-area: title;

		color: $color-beige-dark;

		white-space: pre-line;

		word-wrap: break-word;

		font-family: $font-family-title;
		font-size: $font-size-title-xxlg;

		@media (max-width: $break-md) {
			font-size: $font-size-title-xlg;
			line-height: 4.5rem;
		}
	}

	.author {
		grid-area: author;

		color: $color-beige-dark;

		font-family: $font-family-text;
		font-size: $font-size-title-xs;
		line-height: 3.7rem;

		@media (max-width: $break-md) {
			font-size: $font-size-text-xs;
			line-height: 3.7rem;
		}
	}
}
