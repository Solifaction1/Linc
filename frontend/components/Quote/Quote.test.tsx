import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import Quote from './Quote'
import {
	defaultProps,
	defaultPropsNoRoleAndNoRechtsDomein,
	defaultPropsNoRole,
	defaultPropsNoRechtsDomein,
} from './quote.config'

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}))

describe('Quote', () => {
	it('renders a Quote', () => {
		const { container, getByText } = render(<Quote {...defaultProps} />)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('quote')[0]).toBeVisible()

		expect(container.getElementsByClassName('title')[0]).toBeVisible()
		expect(getByText('title', { selector: 'div' })).toBeVisible()

		expect(container.getElementsByClassName('author')[0]).toBeVisible()
		expect(
			getByText('firstName name - role - title', { selector: 'p' })
		).toBeVisible()
	})

	it('renders a Quote with no role', () => {
		const { container, getByText } = render(<Quote {...defaultPropsNoRole} />)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('quote')[0]).toBeVisible()

		expect(container.getElementsByClassName('title')[0]).toBeVisible()
		expect(getByText('title', { selector: 'div' })).toBeVisible()

		expect(container.getElementsByClassName('author')[0]).toBeVisible()
		expect(getByText('firstName name - title', { selector: 'p' })).toBeVisible()
	})

	it('renders a Quote with no rechtsdomein and no role', () => {
		const { container, getByText } = render(
			<Quote {...defaultPropsNoRoleAndNoRechtsDomein} />
		)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('quote')[0]).toBeVisible()

		expect(container.getElementsByClassName('title')[0]).toBeVisible()
		expect(getByText('title', { selector: 'div' })).toBeVisible()

		expect(container.getElementsByClassName('author')[0]).toBeVisible()
		expect(getByText('firstName name', { selector: 'p' })).toBeVisible()
	})

	it('renders a Quote with no rechtsdomein', () => {
		const { container, getByText } = render(
			<Quote {...defaultPropsNoRechtsDomein} />
		)

		expect(container).toMatchSnapshot()

		expect(container.getElementsByClassName('quote')[0]).toBeVisible()

		expect(container.getElementsByClassName('title')[0]).toBeVisible()
		expect(getByText('title', { selector: 'div' })).toBeVisible()

		expect(container.getElementsByClassName('author')[0]).toBeVisible()
		expect(getByText('firstName name - role', { selector: 'p' })).toBeVisible()
	})
})
