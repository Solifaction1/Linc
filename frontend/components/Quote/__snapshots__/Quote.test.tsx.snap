// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders a Quote 1`] = `
<div>
  <section
    class="quote"
  >
    <div
      class="title"
    >
      title
    </div>
    <p
      class="author"
    >
      firstName name - role - title
    </p>
  </section>
</div>
`;

exports[`Quote renders a Quote with no rechtsdomein 1`] = `
<div>
  <section
    class="quote"
  >
    <div
      class="title"
    >
      title
    </div>
    <p
      class="author"
    >
      firstName name - role
    </p>
  </section>
</div>
`;

exports[`Quote renders a Quote with no rechtsdomein and no role 1`] = `
<div>
  <section
    class="quote"
  >
    <div
      class="title"
    >
      title
    </div>
    <p
      class="author"
    >
      firstName name
    </p>
  </section>
</div>
`;

exports[`Quote renders a Quote with no role 1`] = `
<div>
  <section
    class="quote"
  >
    <div
      class="title"
    >
      title
    </div>
    <p
      class="author"
    >
      firstName name - title
    </p>
  </section>
</div>
`;
